core/src/main/java/jenkins/tasks/SimpleBuildWrapper.java

```diff
@@ -64,18 +66,70 @@
 @SuppressWarnings("rawtypes") // inherited
 public abstract class SimpleBuildWrapper extends BuildWrapper {
 
+    /**
+     * Determines whether or not this wrapper requires a workspace context (working directory and launcher).
+     * <p>
+     * When such a context is required (the default), {@link #setUp(Context, Run, FilePath, Launcher, TaskListener, EnvVars)} applies.
+     * Otherwise, {@link #setUp(Context, Run, TaskListener, EnvVars)} applies.
+     *
+     * @return {@code true} if this wrapper requires a workspace context; {@code false} otherwise.
+     * @since TODO
+     */
+    @Restricted(Beta.class) // to indicate it is to be called by Jenkins internals only; not part of the normal API
```

Comment:
this comment doesn't seem right? beta just means that it's a new API and it may change in some way? To me anyway

Reply:
https://github.com/kohsuke/access-modifier/issues/25

Reply:
It's also why I added the comment, to make it clear its not "regular" `Beta`.

Reply:
```suggestion
```

Wrong here.